crates/electrum/src/electrum_ext.rs

```diff
@@ -1,29 +1,28 @@
 use bdk_chain::{
     bitcoin::{OutPoint, ScriptBuf, Transaction, Txid},
-    keychain::WalletUpdate,
     local_chain::{self, CheckPoint},
     tx_graph::{self, TxGraph},
     Anchor, BlockId, ConfirmationHeightAnchor, ConfirmationTimeAnchor,
 };
 use electrum_client::{Client, ElectrumApi, Error, HeaderNotification};
 use std::{
-    collections::{BTreeMap, BTreeSet, HashMap, HashSet},
+    collections::{BTreeMap, HashMap, HashSet},
     fmt::Debug,
     str::FromStr,
 };
 
 /// We assume that a block of this depth and deeper cannot be reorged.
 const ASSUME_FINAL_DEPTH: u32 = 8;
 
-/// Represents an update fetched from an Electrum server, but excludes full transactions.
+/// Represents an update fetched from an Electrum server.
 ///
-/// To provide a complete update to [`TxGraph`], you'll need to call [`Self::missing_full_txs`] to
+/// To get a complete [`Self::graph_update`], you'll need to call [`Self::missing_full_txs`] to
 /// determine the full transactions missing from [`TxGraph`]. Then call [`Self::finalize`] to fetch
 /// the full transactions from Electrum and finalize the update.
 #[derive(Debug, Clone)]
 pub struct ElectrumUpdate<K, A> {
-    /// Map of [`Txid`]s to associated [`Anchor`]s.
-    pub graph_update: HashMap<Txid, BTreeSet<A>>,
```

Comment:
What is the motivation of this change?

Reply:
The motivation was to get rid of `WalletUpdate` but still have a way to get all the graph updates that we were exposing through `WalletUpdate`.

Reply:
The main motivation for removing `WalletUpdate` from here is so `keychain_update` doesn't have to be part of this struct since it is not always necessary to have it. The one that does the keychain scan should just return a tuple with the keychain update being an element. This removes a type parameter. The `new_tip` also seems orthogonal -- it can just be returned as another element in the tuple. So `ElectrumUpdate` should only have the graph info in it. 

The particular change made here seems a bit dangerous since we can't enforce the user doesn't use the `TxGraph` as an update before it's finalized. If we do what i suggest above I think it's fine to leave this as a `HashMap` and then consume self in the `finalize` methods.